src/Compilers/Core/Portable/SynthesizedLocalKind.cs

```diff
@@ -232,6 +232,12 @@ internal enum SynthesizedLocalKind
         /// TODO: Avoid using lambdas and display classes for implementation of relaxation stubs and remove this kind.
         /// </summary>
         DelegateRelaxationReceiver = 0x101,
+
+        /// <summary>
+        /// The builder type of an interpolated string, when using the interpolated string builder pattern.
+        /// This synthesized local is allowed to have a val escape
+        /// </summary>
+        InterpolatedStringBuilder = 0x102,
```

Comment:
>InterpolatedStringBuilder [](start = 8, length = 25)

Is this just a specialized flavor of ```LoweringTemp```? I.e. it is "allowed to have a val escape" (BTW, it would be good to provde a brief explanation what does that mean exactly). Perhaps we should use a more general name rather than tying it to specific feature and specific use case by name.

Reply:
Yes. I originally had just modified `LoweringTemp`, but got feedback that it might be better as a separate class.

---
In reply to: [610661871](https://github.com/dotnet/roslyn/pull/52422#discussion_r610661871) [](ancestors = 610661871)

Reply:
> Perhaps we should use a more general name rather than tying it to specific feature and specific use case by name.

I like this suggestion, I'll do that.

---
In reply to: [612025077](https://github.com/dotnet/roslyn/pull/52422#discussion_r612025077) [](ancestors = 612025077,610661871)

Reply:
I've created a more general `SynthesizedLocalWithValEscape`, but left this kind as is for ease of debugging.

---
In reply to: [612025148](https://github.com/dotnet/roslyn/pull/52422#discussion_r612025148) [](ancestors = 612025148,612025077,610661871)